model-sync-lib/src/main/kotlin/org/modelix/model/sync/ModelImporter.kt

```diff
@@ -0,0 +1,99 @@
+package org.modelix.model.sync
+
+import org.modelix.model.api.*
+import org.modelix.model.data.ModelData
+import org.modelix.model.data.NodeData
+import java.io.File
+
+class ModelImporter(private val branch: IBranch) {
+
+    private lateinit var originalIdToRef: MutableMap<String, INodeReference>
+
+    fun import(jsonFile: File) {
+        require(jsonFile.exists())
+        require(jsonFile.extension == "json")
+
+        val data = ModelData.fromJson(jsonFile.readText())
+        import(data)
+    }
+    
+    fun import(data: ModelData) {
+        originalIdToRef = mutableMapOf()
+        branch.runWrite {
+            val rootNode = branch.getRootNode()
+            if (rootNode.allChildren.none()) {
```

Comment:
Remove this case. The importer should run its own implementation. `ModelData.load` should probably be removed later and replaced by this importer.